lib/executor.ts

```diff
@@ -36,7 +36,7 @@ export class Executor {
   /**
    * The return value of `currentFn`
    */
-  result?: void | Promise<void>;
+  result?: void | Promise<any>;
```

Comment:
This PR fixes my issue, but I could use some recommendations on how to best type it. Right now I'm just changing all the Promises to be `any`, but it might be better to allow users to pass in the response type of their choice via a generic value. That would be a fairly large refactor to this library though, so I decided to keep it out of this PR for now. Let me know what you think.